flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/NFAITCase.java

```diff
@@ -1391,6 +1391,248 @@ public boolean filter(Event value) throws Exception {
 	}
```

Comment:
Maybe let's move all those new tests into seperate class like `SameElementITCase` or `IterativeConditionsITCase`? The NFAITCase is already too long in my opinion.